lib/src/emitter.dart

```diff
@@ -2,18 +2,30 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'allocator.dart';
+import 'base.dart';
 import 'specs/annotation.dart';
 import 'specs/class.dart';
 import 'specs/code.dart';
 import 'specs/constructor.dart';
+import 'specs/directive.dart';
 import 'specs/field.dart';
+import 'specs/file.dart';
 import 'specs/method.dart';
 import 'specs/reference.dart';
 import 'specs/type_reference.dart';
 import 'visitors.dart';
 
-class DartEmitter extends GeneralizingSpecVisitor<StringSink> {
-  const DartEmitter();
+class DartEmitter implements SpecVisitor<StringSink> {
+  final Allocator _allocator;
```

Comment:
I don't think _allocator.imports is ever called. So how would you actually print out the imports in a file?